packages/apollo-gateway/src/index.ts

```diff
@@ -379,8 +383,11 @@ export class ApolloGateway implements GraphQLService {
     serviceDef: ServiceEndpointDefinition,
   ): GraphQLDataSource {
     // If the DataSource has already been created, early return
-    if (this.serviceMap[serviceDef.name])
-      return this.serviceMap[serviceDef.name];
+    if (
+      this.serviceMap[serviceDef.name] &&
+      serviceDef.url === this.serviceMap[serviceDef.name].url
+    )
+      return this.serviceMap[serviceDef.name].dataSource;
 
     if (!serviceDef.url && !isLocalConfig(this.config)) {
```

Comment:
Should this really throw an error? Nothing that calls this actually handles an error effectively, so that means now that if you push up a bad version of your service with a `service:push`, the gateway will actually just crash rather than failing to update. Can we be more resilient to this instead?

Reply:
Sorry, I realize this isn't new code, but I only just now noticed the strangeness of throwing an error within this function that isn't handled.

Reply:
I haven't traced this precisely, but an unhandled exception in async functions or a rejected promise wouldn't typically result in a crash to the server, though it will certainly prevent the update from happening.  As far as I'm aware, the only current crash currently would be via the initial call to `createSchema` (which eventually arrives here) in the `ApolloGateway` constructor.   I believe the other way that this method is invoked is within async/Promise chains via `load`'s `setInterval`, and would fall into the category of non-terminating errors.

To be clear, a future version of Node.js will actually terminate the server under these conditions, but the brand new Node.js 12 LTS that is meant to arrive next week will still not take this approach

Either way, let's at least handle this now that we're aware of it though.